File: polyglot/vault_file/py/src/vault_file/domain.py
from datetime import datetime, timezone
from typing import Dict, Optional
from pydantic import BaseModel, Field, field_validator
import re

class VaultHeader(BaseModel):
    version: str = Field(default="1.0.0", description="Semantic version of the vault file")
    created_at: str = Field(default_factory=lambda: datetime.now(timezone.utc).isoformat(timespec='milliseconds'), description="ISO 8601 timestamp")
    description: Optional[str] = None

    @field_validator('version')
    def validate_version(cls, v):
        if not re.match(r'^\d+\.\d+\.\d+$', v):
            raise ValueError("Version must be semantic versioning (x.y.z)")
        return v

class VaultFile(BaseModel):
    header: VaultHeader
    secrets: Dict[str, str]

class LoadResult(BaseModel):
    total_vars_loaded: int = Field(alias="totalVarsLoaded")

File: polyglot/vault_file/mjs/src/domain.ts
import { z } from 'zod';

export const VaultHeaderSchema = z.object({
  version: z.string().regex(/^\d+\.\d+\.\d+$/, "Version must be semantic versioning (x.y.z)").default("1.0.0"),
  createdAt: z.string().datetime().default(() => new Date().toISOString()),
  description: z.string().optional(),
});

export type VaultHeader = z.infer<typeof VaultHeaderSchema>;

export const VaultFileSchema = z.object({
  header: VaultHeaderSchema,
  secrets: z.record(z.string()),
});

export type VaultFile = z.infer<typeof VaultFileSchema>;

export interface LoadResult {
  totalVarsLoaded: number;
}
